src/lib.rs

```diff
@@ -87,22 +89,29 @@ pub use ttf_parser::Width as Stretch;
 /// implementation for this type only promise that unequal IDs will be displayed
 /// as different strings, but does not make any guarantees about format or
 /// content of the strings.
+///
+/// [`KeyData`]: https://docs.rs/slotmap/latest/slotmap/struct.KeyData.html
 #[derive(Clone, Copy, Eq, Hash, Ord, PartialEq, PartialOrd, Debug)]
-pub struct ID(u32);
+pub struct ID(InnerId);
+
+slotmap::new_key_type! {
+    /// Internal ID type.
+    struct InnerId;
+}
 
 impl ID {
     /// Creates a dummy ID.
     ///
     /// Should be used in tandem with [`Database::push_face_info`].
     #[inline]
     pub fn dummy() -> Self {
-        Self(0)
+        Self(InnerId::from(slotmap::KeyData::from_ffi(core::u64::MAX)))
```

Comment:
Are you sure this is safe/ok? Is slotmap ID is just an incrementing u64 number?

Reply:
The slotmap ID is just a `u32` representing the index combined with a `u32` representing the current "version" of the index. At worst, it would cause a panic for out-of-bounds access.